Dockerfile

```
@@ -20,9 +20,18 @@ WORKDIR /home/logstash
 # used by the purge policy
 LABEL retention="keep"
 
+# Setup gradle wrapper. When running any `gradle` command, a `settings.gradle` is expected (and will soon be required).
+# This section adds the gradle wrapper, `settings.gradle` and sets the permissions (setting the user to root for `chown`
+# and working directory to allow this and then reverts back to the previous working directory and user.
 ADD gradlew /opt/logstash/gradlew
 ADD gradle/wrapper /opt/logstash/gradle/wrapper
-RUN /opt/logstash/gradlew wrapper
+ADD settings.gradle /opt/logstash/settings.gradle
```

we can replace this ADD with a copy that modified ownership in a single pass to avoid the extra `USER` + `RUN` juggling
```suggestion
COPY --chown=logstash:logstash  settings.gradle /opt/logstash/settings.gradle
```